tasks/main.yml

```diff
@@ -27,19 +27,32 @@
       - gnupg2
       - software-properties-common
 
-- name: Add Docker apt-key
-  ansible.builtin.apt_key:
-    url: https://download.docker.com/linux/debian/gpg
-    state: present
+# Link: https://docs.docker.com/engine/install/debian/#install-using-the-repository
+- name: Prepare apt with key and sources entry
+  block:
+    - name: Ensure apt keyrings dir is present
+      ansible.builtin.file:
+        path: "{{ docker_apt_keyrings_dir }}"
+        state: directory
+        owner: root
+        group: root
+        mode: '0755'
 
-- name: Add Docker's APT repository
-  ansible.builtin.template:
-    src: templates/docker.list.j2
-    dest: /etc/apt/sources.list.d/docker.list
-    owner: root
-    group: root
-    mode: '0644'
-  register: apt_repo
+    # Same file as from https://download.docker.com/linux/debian/gpg
+    - name: Put docker apt release key in place
+      ansible.builtin.copy:
+        src: "files{{ docker_apt_keyrings_dir }}/docker.asc"
+        dest: "{{ docker_apt_keyrings_dir }}"
+        mode: '0644'
```

Comment:
As APT keys are frequently renewed, in an update run there might be a change of keys without a change of repository listing.

To reflect this change in the APT index, `register: apt_repo`is needed here as well.

Reply:
This is addressed in v2, which I just force-pushed.